File: frontend/lib/domain/model/client_settings.dart
import 'package:json_annotation/json_annotation.dart';

part 'client_settings.g.dart';

@JsonSerializable()
class ClientSettings {
  ClientSettings(this.grantTypes, this.redirectUris, this.scope, this.audience,
      this.tokenExpiration, this.refreshTokenExpiration, this.refreshTokenAbsoluteExpiration);

  List<String> grantTypes;
  List<String> redirectUris;
  List<String> scope;
  List<String> audience;
  int tokenExpiration;
  int refreshTokenExpiration;
  int refreshTokenAbsoluteExpiration;

  factory ClientSettings.fromJson(Map<String, dynamic> json) =>
      _$ClientSettingsFromJson(json);

  Map<String, dynamic> toJson() => _$ClientSettingsToJson(
        this,
      );
}

File: backend/src/main/kotlin/app/vatov/idserver/model/ClientSettings.kt
package app.vatov.idserver.model

import kotlinx.serialization.Serializable

@Serializable
data class ClientSettings(
    val grantTypes: List<GrantType> = emptyList(),
    val redirectUris: List<String> = emptyList(),
    val scope: List<String> = emptyList(),
    val audience: List<String> = emptyList(),
    val tokenExpiration: Long = 3600,
    val refreshTokenExpiration: Long = 604800,
    val refreshTokenAbsoluteExpiration: Long = 2592000
)
